contracts/strategies/StrategyDAI3pool.sol

```diff
@@ -0,0 +1,334 @@
+    address public constant ycrv = address(0xdF5e0e81Dff6FAF3A7e52BA697820c5e32D806A8);
+    address public constant yycrv = address(0x5dbcF33D8c2E976c6b560249878e6F1491Bca25c);
+
+    address public constant dai = address(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+    address public constant ydai = address(0x16de59092dAE5CcF4A1E6439D611fd0653f0Bd01);
+    address public constant usdc = address(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+    address public constant yusdc = address(0xd6aD7a6750A7593E092a9B218d66C0A814a3436e);
+    address public constant usdt = address(0xdAC17F958D2ee523a2206206994597C13D831ec7);
+    address public constant yusdt = address(0x83f798e925BcD4017Eb265844FDDAbb448f1707D);
+    address public constant tusd = address(0x0000000000085d4780B73119b644AE5ecd22b376);
+    address public constant ytusd = address(0x73a052500105205d34Daf004eAb301916DA8190f);
+
+    address public governance;
+    address public controller;
+    address public strategist;
+
+    uint256 public constant DENOMINATOR = 10000;
+    uint256 public treasuryFee = 500;
+    uint256 public withdrawalFee = 0;
+    uint256 public strategistReward = 50;
+    uint256 public threshold = 6000;
+    uint256 public slip = 100;
+    uint256 public tank = 0;
+    uint256 public p = 0;
+    bool public flag = true;
+
+    modifier isAuthorized() {
+        require(msg.sender == strategist || msg.sender == governance || msg.sender == controller || msg.sender == address(this), "!authorized");
+        _;
+    }
+
+    constructor(address _controller) public {
+        governance = msg.sender;
+        strategist = msg.sender;
+        controller = _controller;
+    }
+
+    function getName() external pure returns (string memory) {
+        return "StrategyDAI3pool";
+    }
+
+    function unwindHelper() internal {
+        uint256 _want = IERC20(want).balanceOf(address(this));
+        IERC20(want).safeApprove(ypool, 0);
+        IERC20(want).safeApprove(ypool, _want);
+
+        _want = _want.div(3);
+        ICurveFi(ypool).exchange_underlying(0, 1, _want, 0);
+        ICurveFi(ypool).exchange_underlying(0, 2, _want, 0);
+
+        uint256 usdc_amt = IERC20(usdc).balanceOf(address(this));
+        IERC20(usdc).safeApprove(_3pool, 0);
+        IERC20(usdc).safeApprove(_3pool, usdc_amt);
+
+        uint256 usdt_amt = IERC20(usdt).balanceOf(address(this));
+        IERC20(usdt).safeApprove(_3pool, 0);
+        IERC20(usdt).safeApprove(_3pool, usdt_amt);
+
+        uint256 dai_amt = IERC20(dai).balanceOf(address(this));
+        IERC20(dai).safeApprove(_3pool, 0);
+        IERC20(dai).safeApprove(_3pool, dai_amt);
+
+        uint256 v = _want.mul(1e18).div(ICurveFi(_3pool).get_virtual_price());
+        ICurveFi(_3pool).add_liquidity([dai_amt, usdc_amt, usdt_amt], v.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR));
+    }
+
+    function deposit() public isAuthorized {
+        if (flag) {
```

Comment:
this is a one time shot according to the code on migration, i assume there is no scenario where we ever could need to turn back on this after first deposit/migration from strategy?

Reply:
exactly. this `flag` is only for the one-time unwinding migration. and then it could start working with auto-rebalancing like others.